Add new `Discount banner` pattern

albarin opened the pull request:
This PRs adds the new `Discount banner` pattern.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9930

### Screenshots
<img width="481" alt="Screenshot 2023-06-21 at 15 31 03" src="https://user-images.githubusercontent.com/186112/247502460-9ff3235a-7770-4ff4-ac58-ef533737731e.png">

### Testing

#### User-Facing Testing

1. Create a new post or page.
2. Insert the `Discount banner` pattern and save.
3. In the front end, check the pattern looks like the image above and the `Shop now` button links to the shop page.

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Add the new `Banner discount` pattern.

albarin committed:
Add new `Discount banner` pattern

albarin committed:
Add shop link

kmanijak commented on patterns/discount-banner.php:
```diff
@@ -0,0 +1,42 @@
+		<!-- /wp:paragraph -->
+
+		<!-- wp:paragraph {"style":{"typography":{"fontStyle":"normal","fontWeight":"300","fontSize":"35px"}},"textColor":"base"} -->
+		<p class="has-base-color has-text-color" style="font-size:35px;font-style:normal;font-weight:300">Select product</p>
```
Minor, but design says "Select productS":

```suggestion
		<p class="has-base-color has-text-color" style="font-size:35px;font-style:normal;font-weight:300">Select products</p>
```

kmanijak reviewed (changes requested):
Looks good!

I checked the banner on various styles and found one caveat. On Canary (TT3 theme) the font is a big bigger and there's not enough width for "40% off". And because of the `line-height: 0` it's stack like in the image:

<img width="488" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/5a314553-e4c5-4658-98c0-1a12281734b1">

But I understand it's opinionated and should work well in MOST cases, not EVERY case. Hence just mentioning it, but I don't think that's a must.

Also, on designs all the text is left-aligned, while "40% off" seems to have whitespace around it.
<img width="103" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/9cc3fe2d-7618-4ee2-8352-48c5b150ae20">

<img width="491" alt="Screen Shot 2023-06-22 at 08 56 00 AM" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/a0ce4922-ca5d-4fcb-a96b-f79e10e5ea9d">

Removing them could actually help with the text wrapping in Canary!

albarin committed:
Fix copy

albarin committed:
Fix colors

albarin commented:
Thanks for the review @kmanijak! I've fixed your comments and also added some hardcoded colors that were missing.
Looking better on Canary and a few more themes I tried. Could you have another look?
<img width="537" alt="Screenshot 2023-06-22 at 09 52 34" src="https://github.com/woocommerce/woocommerce-blocks/assets/186112/cf2c89ce-c037-4602-9ef5-5a636f03b580">

kmanijak reviewed (approved):
Looks great!
